pkg/azure/externalmetrics/monitor_client.go

```diff
@@ -55,23 +55,33 @@ func (c *monitorClient) GetAzureMetric(azMetricRequest AzureExternalMetricReques
 		return AzureExternalMetricResponse{}, err
 	}
 
-	total := extractValue(metricResult)
-
-	klog.V(2).Infof("found metric value: %f", total)
+	value := extractValue(azMetricRequest, metricResult)
 
 	// TODO set Value based on aggregations type
 	return AzureExternalMetricResponse{
-		Total: total,
+		Value: value,
 	}, nil
 }
 
-func extractValue(metricResult insights.Response) float64 {
-	//TODO extract value based on aggregation type
-	//TODO check for nils
```

Comment:
Since you are modifying this section of code, could you add some nill checks to help improve?